client/src/components/layout/CommentForm.js

```diff
@@ -0,0 +1,107 @@
+import React, { useState, useEffect } from "react"
+import getCurrentUser from "../../services/getCurrentUser.js"
+
+import translateServerErrors from "../../services/translateServerErrors.js"
+import ErrorList from "./ErrorList.js"
+
+const Comment = (props) => {
+  const [comment, setComment] = useState({
+    title: "",
+    content: "",
+    userId: "",
+    tripId: "",
+  })
+  const [errors, setErrors] = useState([])
+
+  const [currentUser, setCurrentUser] = useState({
+    id: "",
+    email: "",
+    userName: "",
+  })
+  useEffect(() => {
+    getCurrentUser()
+      .then((user) => {
+        setCurrentUser(user)
+      })
+      .catch(() => {
+        setCurrentUser(null)
+      })
+  }, [])
+
+  comment.tripId = props.tripId
+  comment.userId = currentUser.id
+
```

Comment:
this is a HUGE no-no. Here you effectively changing state _without_ using `setComment`

if you want the tripId and userId, create a new object in your `handlePost` request that has the same properties as `comment` plus tripId and userId. the spread operator would help

```
let fullComment = {
  ...comment,
  tripId,
  userId
}
```
```

Reply:
the above doesnt edit state, but instead makes a new object with the same properties as `comment` plus your ids